com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/DebugAdapter.java

```diff
@@ -72,8 +72,7 @@ public DebugAdapter(IProtocolServer server, IProviderContext providerContext) {
         Command command = Command.parse(request.command);
         Arguments cmdArgs = JsonUtils.fromJson(request.arguments, command.getArgumentType());
 
-        if (debugContext.isVmTerminated()) {
-            // the operation is meaningless
+        if (debugContext.isVmTerminated() && command != Command.DISCONNECT) {
```

Comment:
curious about what does `command != Command.DISCONNECT` fix?

Reply:
Don't intercept disconnect request here because we want to leave it to disconnectHandler to cleanup temp launch files.